rbd/pool_nautilus.go

```diff
@@ -98,3 +98,27 @@ func PoolInit(ioctx *rados.IOContext, force bool) error {
 	ret := C.rbd_pool_init(cephIoctx(ioctx), C.bool(force))
 	return getError(ret)
 }
+
+// PoolStats represents RBD pool stats variable.
+type PoolStats struct {
+	stats C.rbd_pool_stats_t
+}
+
+// PoolStatsCreate a new PoolStats struct.
+// Call PoolStats.Destroy to free the resources.
+//
+// Implements:
+//  void rbd_pool_stats_create(rbd_pool_stats_t *stats)
+func PoolStatsCreate() *PoolStats {
+	poolStats := &PoolStats{}
+	C.rbd_pool_stats_create(&poolStats.stats)
+	return poolStats
+}
+
+// Destroy a PoolStats struct and free the associated resources.
+//
+// Implements:
+//  void rbd_pool_stats_destroy(rbd_pool_stats_t stats)
+func (poolStats *PoolStats) Destroy() {
+	C.rbd_pool_stats_destroy(poolStats.stats)
```

Comment:
consider setting `poolStats.stats` to `nil` to prevent problematic usage after destroying.

Reply:
Added

Reply:
now I wonder what happens when you call `poolStat.Destroy()` 2x :wink:

Reply:
I guess that's fine, I have a testcase where I am calling poolStats.Destroy()

Reply:
did you try calling it twice? some of the Ceph library functions crash when getting called with a `null` argument.

Reply:
But why some one should call Destroy() twice? Its like calling free() twice

Reply:
We should not, but users could. Adding a check for `poolStats.stats != nil` makes it less likely that users get a segfault on incorrect API usage.

Reply:
Added a check